src/Cursors.ts

```diff
@@ -3,34 +3,82 @@ import { Types } from 'ably';
 import Space from './Space.js';
 import CursorBatching from './CursorBatching.js';
 import CursorDispensing from './CursorDispensing.js';
-import EventEmitter, {
-  InvalidArgumentError,
-  inspect,
-  type EventKey,
-  type EventListener,
-} from './utilities/EventEmitter.js';
+import EventEmitter, { InvalidArgumentError, inspect, type EventListener } from './utilities/EventEmitter.js';
 import CursorHistory from './CursorHistory.js';
 import { CURSOR_UPDATE } from './CursorConstants.js';
 
 import type { CursorsOptions, CursorUpdate } from './types.js';
 import type { RealtimeMessage } from './utilities/types.js';
 import { ERR_NOT_ENTERED_SPACE } from './Errors.js';
 
-type CursorsEventMap = {
+/**
+ * The property names of `CursorsEventMap` are the names of the events emitted by { @link Cursors }.
+ */
+export interface CursorsEventMap {
+  /**
+   * A space member moved their cursor.
+   */
   update: CursorUpdate;
-};
+}
 
 const CURSORS_CHANNEL_TAG = '::$cursors';
 
+/**
+ * > **Documentation source**
+ * >
+ * > The following documentation is copied from the [Spaces documentation website](https://ably.com/docs/spaces).
```

Comment:
I'm somewhat uneasy about these big chunks of documentation being copied from docs. These are bound to get very quickly out of sync (in fact - they already are as we now have a textile and markdown version of the same docs). Is this how it is done in other SDKs or a normal practice that I'm unaware of? Maybe we could have a summary here and link to `/docs/spaces`?

@m-hulbert any thoughts on this?

Reply:
For the Ably client SDKs, the tech writers have written dedicated documentation comments. See https://github.com/ably/sdk-api-reference/blob/main/descriptions.md. The Ably client SDKs (e.g. ably-js, ably-cocoa) have then copied and pasted these descriptions and made some platform-specific edits.

Since we don't have the same thing in Spaces (high-quality docstrings written by tech writers) I instead tried to make use of whichever tech writer-written documentation we did have, hence the copying and pasting from the docs repo. (This documentation was in general of a higher quality that that in `class-definitions.md`.) I do think that if we had time then it would be good to do the same thing as our other SDKs — that is, ask Mark to write proper docstrings for us to use.

As for whether — if we decide to make use of the content in the docs repo — we should be linking to it or copying-and-pasting it, I don't know. I thought we wanted these documentation comments to be useful by themselves, without needing a user to navigate to a separate website. But it's probably a question for @Srushtika.

Reply:
I see, thanks for explaining @lawrence-forooghian. It sounds this falls under what we discussed in our stand-ups, and this is the part that @m-hulbert and the team can update over time. I was thrown off by the comment `The following documentation is copied from the [...]` as it suggested to me that there is supposed to be a sync of some kind, but I can see that's not the intent here (or rather the decision is going to be with the docs team).

Reply:
however, I think the comment, which I can now see is visible on https://sdk.ably.com/builds/ably/spaces/pull/204/typedoc/ is I think unnecessary. The fact the docs are copied, at this point in time, is I think irrelevant to the user and is more for our information. Could we remove them please
![Screenshot 2023-10-16 at 14 09 12](https://github.com/ably/spaces/assets/1148118/aff58036-0000-4b89-883c-5e1ca0b36b8e)

Reply:
I did that because I wanted it to be clear to users that the documentation hadn't been properly edited yet and might not make full sense in this new context. Also, there are places where we have both website documentation and `class-definitions.md` documentation and it doesn't read well if you just put one after another without anything to separate them.

The correct solution is to edit the documentation so that none of this stuff is needed, but I think that in the meantime it makes sense to keep these things there to make it clear that we know that the documentation might not be great. I am happy to remove all of these markers though if you prefer. What do you think?

Reply:
(I do agree that the text "The following documentation was copied from the former `docs/class-definitions.md` document." is quite nonsensical though — why would a user care about a document that no longer exists...)

Reply:
I'd for removing them - we're not planning to cut a release yet so there should be plenty of time for the docs team to review.

Reply:
OK, cool, will do.

Reply:
Done!